task05/server.js

```diff
@@ -0,0 +1,46 @@
+const http = require('http');
+const {
+  getUsers,
+  getUserHobbies,
+  createUser,
+  updateUserHobbies,
+  removeUser,
+} = require('./controllers/userController');
+
+const url = '/api/users';
+const urlWithUserId = `${url}/[a-zA-Z0-9-]+`;
+
+const server = http.createServer(async (req, res) => {
+  const pathName = req.url;
+
+  if (pathName === url && req.method === 'GET') {
+    await getUsers(req, res);
+  } else if (pathName === url && req.method === 'POST') {
+    await createUser(req, res);
+  } else if (
+    req.url.match(new RegExp(`${urlWithUserId}`)) &&
+    req.method === 'DELETE'
+  ) {
+    const id = req.url.split('/')[3];
+    await removeUser(req, res, id);
+  } else if (
+    req.url.match(new RegExp(`${urlWithUserId}\/hobbies`)) &&
+    req.method === 'GET'
+  ) {
+    const id = req.url.split('/')[3];
+    await getUserHobbies(req, res, id);
+  } else if (
+    req.url.match(new RegExp(`${urlWithUserId}\/hobbies`)) &&
+    req.method === 'PATCH'
+  ) {
+    const id = req.url.split('/')[3];
+    await updateUserHobbies(req, res, id);
+  } else {
+    res.writeHead(404, { 'Content-Type': 'application/json' });
+    res.end(JSON.stringify({ message: "This route doesn't exist" }));
+  }
```

Comment:
It is best to avoid such massive designs, as there will be additional costs involved to add or modify anything in the future.

Personally, I would go with a scenario of an object and a router resolver (which gets the configuration and requester by comparing all the data internally):

```javascript
const routesConfig = [
  { path: '/api/users', method: 'GET', handler: getUsers },
  { path: '/api/users', method: 'POST', handler: createUser },
  // ...
];
```

This approach will allow you to add new paths very quickly, also consider that the logic related to receiving the request parameters is usually also the responsibility of the controller.